#160451599 User should be able to update his Profile

obulaworld opened the pull request:
#### What does this PR do?
- Updates user's profile
   
#### Description of Task to be completed?
  - Implement update user profile API ````PUT /api/v1/users/profile/:id````
  - Implement view user profile  API ````GET /api/v1/users/profile/:id````
  - Deny user's access to update other user's account 

#### How should this be manually tested?
    - Clone the repo
    - cd into the 'lotus-ah'
    - Run 'npm install'
    - Use the .env.sample format to create a .env file
    - Run 'npm run migrate'
    - Run 'npm run start:dev'

#### Any background context you want to provide?
NA

#### What are the relevant pivotal tracker stories?
[#160451599](https://www.pivotaltracker.com/story/show/160451599 "update user profile")
 
### Question
 NA

### Screenshot
 NA

andela-phemmy committed:
ft(auth): configured and initialized social auth

andela-phemmy committed:
ft(api): Enabled social authentication

 - Configured passport integration in config/passport.js file
 - Created social authentication route
 - Created controller for social auth in controllers/SocialAuthController
 - Initialized passport in index.js

feat(Auth): implemented social authentication

 - Created a PassportMockStrategy to mock response
 - Implemented the passport for social authentication

Delivers #161081708

fix: implement TTL feedback

 - align Jsdoc properly
 - add new line to eslintignore
 - removed unused line from index
  - remove blank lines

obulaworld committed:
feat(user profile) User should be able to update Profile
User should be able to update his profile
User should be able to view other user's profile
User shoulld not be able to update other user's profile
closes 160451599

obulaworld committed:
feat(user profile) User should be able to update Profile)
User should be able to update his profile
User should be able to view other user's profile
User shoulld not be able to update other user's profile
closes 160451599

obulaworld committed:
feat(user profile) User should be able to update Profile)
User should be able to update his profile
User should be able to view other user's profile
User shoulld not be able to update other user's profile
closes 160451599

koiic reviewed (approved):
Lookin good to me

obulaworld committed:
feat(user profile) User should be able to update Profile
User should be able to update his profile
User should be able to view other user's profile

closes 160451599

obulaworld committed:
feat(user profile) User should be able to update Profile
User should be able to update his profile
User should be able to view other user's profile

closes 160451599

KaiserPhemi committed:
Merge pull request #21 from andela/ft-enable-social-login

#111504508 Enable social authentication (AUTH)

KaiserPhemi reviewed (changes requested):
#### Critical
Fix all merge conflicts.

obulaworld committed:
feat(user profile) User should be able to update Profile
User should be able to update his profile
User should be able to view other user's profile

closes 160451599

obulaworld committed:
feat(user profile) User should be able to update Profile
User should be able to update his profile
User should be able to view other user's profile

closes 160451599
